cparser/Cutil.ml

```diff
@@ -533,6 +533,28 @@ let sizeof_struct env members =
       end
   in sizeof_rec 0 members
 
+(* Compute the offset of a struct member *)
+let offsetof env ty fields =
+  let rec sub acc name = function
+    | [] -> List.rev acc
+    | m::rem -> if m.fld_name = name then
+        List.rev acc
+      else
+        sub (m::acc) name rem in
+  let offset (ofs,ty) field =
+    match unroll env ty with
+    | TStruct (id,_) ->
+      let str = Env.find_struct env id in
+      let pre = sub [] field.fld_name str.ci_members in
+      begin match sizeof_struct env pre ,alignof env field.fld_typ with
+      | Some s, Some a ->
+        (ofs + align s a),field.fld_typ
```

Comment:
I started implementing a new simpler version today which does exactly this. One of the problems of this implementation is that it can not handle array designators. But at the moment it is not ready to be committed.